PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/serviceexplorer/NodeActionsMap.java

```diff
@@ -62,6 +63,9 @@
             new HashMap<>();
 
     static {
+        // enable spark serverless node subscribe actions
```

Comment:
Can we change our codes to work like the following implementations?

Reply:
we cannot implement our codes like
```java
    node2Actions.put(SparkServerlessClusterRootModule.class,
            new ImmutableList.Builder<Class<? extends NodeActionListener>>()
                    .add(SparkServerlessClusterOpsCtrl.class).build());
```
Since the value of the map is required to be of Class<? extends NodeActionListener>> type.
However, SparkServerlessClusterOpsCtrl is not of the required type.

Maybe I can leverage the ServiceManager [in this PR](https://github.com/Microsoft/azure-tools-for-java/pull/1623/files) that you mentioned before ?

Reply:
Looks wield to initiate SparkServerlessClusterOpsCtrl here.

---
In reply to: [186369832](https://github.com/Microsoft/azure-tools-for-java/pull/1625#discussion_r186369832) [](ancestors = 186369832)

Reply:
Offline with Wei. I will try to rewrite the codes to align with the native codes.